taca/utils/bioinfo_tab.py

```diff
@@ -26,59 +26,43 @@ def __init__(self, value=None):
 
 def collect_runs():
     """Update command."""
-    found_runs = {"illumina": [], "element": []}
+
     # Pattern explained:
     # 6-8Digits_(maybe ST-)AnythingLetterornumberNumber_Number_AorBLetterornumberordash
     illumina_rundir_re = re.compile("\d{6,8}_[ST-]*\w+\d+_\d+_[AB]?[A-Z0-9\-]+")
+    # E.g. 20250121_AV242106_B2425434199
+    element_rundir_re = re.compile("\d{8}_AV242106_[AB]\d+")
+
     for inst_brand in CONFIG["bioinfo_tab"]["data_dirs"]:
         for data_dir in CONFIG["bioinfo_tab"]["data_dirs"][inst_brand]:
             if os.path.exists(data_dir):
                 potential_run_dirs = glob.glob(os.path.join(data_dir, "*"))
-                for run_dir in potential_run_dirs:
-                    if os.path.isdir(run_dir):
-                        if inst_brand == "illumina" and illumina_rundir_re.match(
-                            os.path.basename(os.path.abspath(run_dir))
-                        ):
-                            found_runs[inst_brand].append(os.path.basename(run_dir))
-                            logger.info(f"Working on {run_dir}")
-                            update_statusdb(run_dir, inst_brand)
-                        elif inst_brand == "element":
-                            # Skip no sync dirs, they will be checked below
-                            if run_dir == os.path.join(data_dir, "nosync"):
-                                continue
-                            logger.info(f"Working on {run_dir}")
-                            update_statusdb(run_dir, inst_brand)
-                        elif inst_brand == "ont":
-                            # Skip archived, no_backup, nosync and qc folders
-                            if re.match(
-                                ONT_RUN_PATTERN,
-                                os.path.basename(os.path.abspath(run_dir)),
-                            ):
-                                logger.info(f"Working on {run_dir}")
-                                update_statusdb(run_dir, inst_brand)
+                potential_run_dirs += glob.glob(os.path.join(data_dir, "nosync", "*"))
 
-                nosync_data_dir = os.path.join(data_dir, "nosync")
-                potential_nosync_run_dirs = glob.glob(
-                    os.path.join(nosync_data_dir, "*")
-                )
-                for run_dir in potential_nosync_run_dirs:
+                for run_dir in potential_run_dirs:
                     if os.path.isdir(run_dir):
                         if (
-                            inst_brand == "illumina"
-                            and illumina_rundir_re.match(
-                                os.path.basename(os.path.abspath(run_dir))
+                            (
+                                inst_brand == "illumina"
+                                and illumina_rundir_re.match(os.path.basename(run_dir))
                             )
-                        ) or (inst_brand == "element" or inst_brand == "ont"):
```

Comment:
The bug is in this block, it only skips processing for "archived", causing it to encounter an error when trying to instantiate an ONT run from the "no_backup" dir

Reply:
Right, and it's fixed by always checking against the ONT_RUN_PATTERN before updating statusdb? 👍

Reply:
Yes, now it will only act on run dirs whose name matches the pattern of the instrument type, we've essentially moved from a blacklist to a whitelist approach.